tests/fixtures/synthetic_hf_state.py

```diff
@@ -0,0 +1,100 @@
+from collections.abc import Iterable
+
+import pytest
+import torch
+from transformers import PreTrainedTokenizer
+
+from composer.core.state import State
+from composer.datasets.dataloader import DataLoaderHparams
+from composer.datasets.lm_datasets import LMDatasetHparams
+from composer.datasets.synthetic_lm import generate_synthetic_tokenizer, synthetic_hf_dataset_builder
+from composer.models import BERTHparams, GPT2Hparams, TransformerHparams
+from composer.models.transformer_shared import ComposerTransformer
+from tests.common.models import generate_dummy_model_config
+from tests.datasets import test_synthetic_lm_data
+
+
+def make_dataset_configs():
+    lm_dataset_configs = [
+        config[0] for config in test_synthetic_lm_data.generate_parameter_configs(
+            ['num_samples', 'chars_per_sample', 'column_names', 'tokenizer_family'])
+    ]
+    for config in lm_dataset_configs:
+        config['drop_last'] = False
+        config['use_masked_lm'] = config['tokenizer_family'] == 'bert'
+    return lm_dataset_configs
+
+
+def make_lm_tokenizer(config: dict):
+    dataset = synthetic_hf_dataset_builder(num_samples=config['num_samples'],
+                                           chars_per_sample=config['chars_per_sample'],
+                                           column_names=config['column_names'])
+    tokenizer = generate_synthetic_tokenizer(config['tokenizer_family'], dataset)
+    return tokenizer
+
+
+def make_dummy_lm(model_name: str, max_position_embeddings: int, tokenizer: PreTrainedTokenizer):
+    pytest.importorskip("transformers")
```

Comment:
perhaps leave pytest skipping control to the invoking test? probably don't want a helper function silently skipping a test.